ratis-netty/src/main/java/org/apache/ratis/netty/server/DataStreamManagement.java

```diff
@@ -224,10 +230,29 @@ StreamInfo remove(ClientInvocationId key) {
     }
   }
 
+  public static class RequestTracker {
+    private final Map<ChannelId, Set<ClientInvocationId>> ref = new ConcurrentHashMap<>();
+
+    public void add(ChannelId channelId,
+                    ClientInvocationId clientInvocationId) {
+      ref.computeIfAbsent(channelId, (e) -> new CopyOnWriteArraySet<>()).add(clientInvocationId);
```

Comment:
`ConcurrentHashMap` is more efficient than `CopyOnWriteArraySet`.  Let's declare it as 
```java
    private final Map<ChannelId, Map<ClientInvocationId, ClientInvocationId>> map = new ConcurrentHashMap<>();
```
and use `ConcurrentHashMap`.